src/framework/execution/types/validation.ts

```diff
@@ -1,6 +1,11 @@
-import { IntegrationExecutionContext } from './context';
+import { ExecutionContext, IntegrationExecutionContext } from './context';
 import { IntegrationInstanceConfig } from './instance';
 
-export type InvocationValidationFunction<
+export type InvocationValidationFunction<T extends ExecutionContext> = (
+  context: T,
+) => Promise<void> | void;
+
+// TODO: This shouldn't be necessary anymore
```

Comment:
It's unclear to me what isn't necessary anymore, nor what needs to be done about it 😄